### PR TITLE
bump `@vue/tsconfig` package to ^0.8.0

### PR DESCRIPTION
**Pull request by TaranaKhanna**
Cherry-picked commit from Renovate PR  #1065  to update @vue/tsconfig to ^0.8.0.

## Summary by CodeRabbit

* **Chores**
  * Bumped a development tooling devDependency to a newer minor version.

* **Bug Fixes / Stability**
  * Added defensive guards across dashboard stores, posts, boards, roadmaps, roles, slot rendering, and route parameter handling to avoid runtime errors in edge cases.

* **Tests**
  * Made event-emission assertions more resilient to absent payloads.

* No user-facing UI changes.

### CODE REVIEW

**Comment by TaranaKhanna** [comment]
@mittalyashu Ready for review.

**Comment by Ja4V8s28Ck** [comment]
The changes in the #1065 and this PR looks the same. Any particular reason on why you made this PR? The `Test/Theme/Build` check is failing, can you take a look at it?

**Comment by Ja4V8s28Ck** [comment]
@TaranaKhanna Are you sure the those formatting's are causing the error?

File: packages/theme/src/pages/posts/_slug/Edit.vue [hunk 1]
```diff
@@ -121,7 +103,11 @@ async function getPost() {
   postLoading.value = true;
 
   const route = router.currentRoute.value;
-  const slug = route.params.slug.toString();
+  const slug = route.params.slug?.toString();
+
+  if (!slug) {
+    return;
+  }
```
Comment (mittalyashu) [inline packages/theme/src/pages/posts/_slug/Edit.vue:129]: @TaranaKhanna can you make this inline?

File: packages/theme/src/pages/posts/_slug/Edit.vue [hunk 2]
```diff
@@ -2,37 +2,19 @@
...
+        @hide-error="hideTitleError" />
+      <l-textarea :model-value="post.contentMarkdown ?? undefined"
+        @update:model-value="(value) => post.contentMarkdown = value ?? null" label="Description"
+        name="Post description" placeholder="What would you use it for?" :disabled="updatePostPermissionDisabled" />
+      <div class="flex justify-start">
+        <Button type="primary" :loading="postSubmitting" :disabled="updatePostPermissionDisabled" @click="savePost">
+          Update
+        </Button>
+      </div>
+    </div>
```
Comment (mittalyashu) [inline packages/theme/src/pages/posts/_slug/Edit.vue]: @TaranaKhanna fix the code formatting

File: packages/theme/src/pages/boards/_url.vue [hunk 3]
```diff
@@ -5,19 +5,13 @@
...
               <template #icon>
                 <sort-desc-icon />
               </template>
               Latest
             </tab-item>
-            <tab-item
-              :active="tab === 'oldest'"
-              @click="updateTab('oldest')"
-            >
+            <tab-item :active="tab === 'oldest'" @click="updateTab('oldest')">
```
Comment (mittalyashu) [inline packages/theme/src/pages/boards/_url.vue]: @TaranaKhanna keep this code as multi-line.

**Review by mittalyashu — CHANGES_REQUESTED** [review]

File: packages/theme/src/pages/posts/_slug/Edit.vue (hunk 1 above)
Comment (TaranaKhanna) [reply packages/theme/src/pages/posts/_slug/Edit.vue:129]: yes

**Comment by TaranaKhanna** [comment]
@mittalyashu  ready for review.

File: packages/theme/src/ee/store/dashboard/boards.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export const useDashboardBoards = defineStore("dashboardBoards", () => {
     );
     if (boardIdx === -1) return;
 
-    Object.assign(boards.value[boardIdx], board);
+    if (boards.value[boardIdx]) Object.assign(boards.value[boardIdx], board);
```
Comment (mittalyashu) [inline packages/theme/src/ee/store/dashboard/boards.ts]: Just like we did an early return statement for `boardIdx`.

Can you do the same for `boards.value[boardIdx]`?

File: packages/theme/src/ee/store/dashboard/roadmaps.ts [hunk 5]
```diff
@@ -63,7 +63,8 @@ export const useDashboardRoadmaps = defineStore("dashboardRoadmaps", () => {
     );
     if (roadmapIdx === -1) return;
 
-    Object.assign(roadmaps.value[roadmapIdx], roadmap);
+    if (roadmaps.value[roadmapIdx])
+      Object.assign(roadmaps.value[roadmapIdx], roadmap);
```
Comment (mittalyashu) [inline packages/theme/src/ee/store/dashboard/roadmaps.ts]: Same as [this](https://github.com/logchimp/logchimp/pull/1447/files#r2506840990)

File: packages/theme/src/ee/store/dashboard/roles.ts [hunk 6]
```diff
@@ -40,7 +40,7 @@ export const useDashboardRoles = defineStore("dashboardRoles", () => {
     const roleIdx = roles.value.findIndex((item) => item.id === role.id);
     if (roleIdx === -1) return;
 
-    Object.assign(roles.value[roleIdx], role);
+    if (roles.value[roleIdx]) Object.assign(roles.value[roleIdx], role);
```
Comment (mittalyashu) [inline packages/theme/src/ee/store/dashboard/roles.ts]: Same as [this](https://github.com/logchimp/logchimp/pull/1447/files#r2506840990)

File: packages/theme/src/store/dashboard/posts.ts [hunk 7]
```diff
@@ -55,7 +55,7 @@ export const useDashboardPosts = defineStore("dashboardPosts", () => {
     );
     if (postIdx === -1) return;
 
-    Object.assign(posts.value[postIdx], post);
+    if (posts.value[postIdx]) Object.assign(posts.value[postIdx], post);
```
Comment (mittalyashu) [inline packages/theme/src/store/dashboard/posts.ts]: Same as [this](https://github.com/logchimp/logchimp/pull/1447/files#r2506840990)

File: packages/theme/src/pages/boards/_url.vue [hunk 8]
```diff
@@ -96,7 +96,13 @@ function updateTab(tabValue: string) {
 
 async function getBoard() {
   loading.value = true;
-  const url = route.params.url.toString();
+  const url = route.params.url?.toString();
+
+  if (!url) {
```
Comment (mittalyashu) [inline packages/theme/src/pages/boards/_url.vue]:
```suggestion
  const url = (route.params.url || "").toString();
  if (!url) {
```

File: packages/theme/src/pages/posts/_slug/Edit.vue [hunk 9]
```diff
@@ -121,7 +121,9 @@ async function getPost() {
   postLoading.value = true;
 
   const route = router.currentRoute.value;
-  const slug = route.params.slug.toString();
+  const slug = route.params.slug?.toString();
+
+  if (!slug) return;
```
Comment (mittalyashu) [inline packages/theme/src/pages/posts/_slug/Edit.vue]:
```suggestion
  const slug = (route.params.slug || "").toString();
  if (!slug) return;
```

Comment (mittalyashu) [reply packages/theme/src/pages/posts/_slug/Edit.vue]: Can you also add return in if `!slug` condition.

- `isPostExist`
- `postLoading`

**Review by mittalyashu — CHANGES_REQUESTED** [review]

File: packages/theme/src/pages/posts/_slug/Edit.vue (hunk 9 above)
Comment (TaranaKhanna) [reply packages/theme/src/pages/posts/_slug/Edit.vue]: ok

**Comment by TaranaKhanna** [comment]
Ready for review, @mittalyashu

**Comment by mittalyashu** [comment]
LGTM.

Had to make couple of patches.
